inference-engine/tests/functional/plugin/cpu/single_layer_tests/topk.cpp

```diff
@@ -0,0 +1,189 @@
+        inputs.insert({ funcInputs[1].get_node_shared_ptr(), kTensor });
+    }
+
+    void SetUp() override {
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+        ngraph::element::Type inputPrecision;
+        ShapeParams shapes;
+        std::int64_t axis;
+        std::int64_t k;
+        ngraph::opset4::TopK::Mode mode;
+        ngraph::opset4::TopK::SortType sortType;
+        std::tie(inputPrecision, shapes, axis, k, mode, sortType) = this->GetParam();
+
+        inputDynamicShapes = { shapes.dataShape.first, shapes.kShape.first };
+        for (size_t i = 0; i < shapes.dataShape.second.size(); ++i) {
+            targetStaticShapes.push_back({ shapes.dataShape.second[i], shapes.kShape.second[i] });
+        }
+
+        auto params = ngraph::builder::makeDynamicParams(inputPrecision, { inputDynamicShapes[0] });
+        auto k_param = std::make_shared<ngraph::opset1::Parameter>(ngraph::element::i32, inputDynamicShapes[1]);
+        params.push_back(k_param);
+        auto topk = std::make_shared<ngraph::opset1::TopK>(params[0], k_param, axis, mode, sortType);
+
+        ngraph::ResultVector results;
+        for (size_t i = 0; i < topk->get_output_size(); ++i) {
+            results.push_back(std::make_shared<ngraph::opset1::Result>(topk->output(i)));
+        }
+
+        function = std::make_shared<ngraph::Function>(results, params, "TopKLayerCPUTest");
+        functionRefs = ngraph::clone_function(*function);
+    }
+};
+
+TEST_P(TopKLayerCPUTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    run();
+    // TODO: Should be uncommented after updating the CheckPluginRelatedResults() method
+    //CheckPluginRelatedResults(executableNetwork, "TopK");
+}
+
+const ngraph::element::TypeVector inputPrecisions = {
+    ngraph::element::f32,
+};
+
+const std::vector<int64_t> axes = { 0, 1, 2 };
+
+const std::vector<int64_t> k = { 1, 3, 5 };
+
+const std::vector<ngraph::opset4::TopK::Mode> modes = {
+    ngraph::opset4::TopK::Mode::MIN,
+    ngraph::opset4::TopK::Mode::MAX
+};
+
+const std::vector<ngraph::opset4::TopK::SortType> sortTypes = {
+    ngraph::opset4::TopK::SortType::SORT_INDICES,
+    ngraph::opset4::TopK::SortType::SORT_VALUES,
+};
+
+const std::vector<ShapeParams> inShapes = {
+    ShapeParams{
+        InputShape{
+            // dynamic
+            {-1, -1, -1},
+            // target
+            {
+                {10, 13, 11},
+                {15, 10, 5},
+                {5, 5, 8}
+            }
+        },
+        // the shape for k input is always scalar
+        InputShape{{}, {{}, {}, {}}}
+    },
+    ShapeParams{
+        InputShape{
+            // dynamic
+            {-1, { 7, 25 }, -1},
+            // target
+            {
+                {10, 7, 10},
+                {15, 20, 5},
+                {5, 25, 7}
+            }
+        },
+        // the shape for k input is always scalar
+        InputShape{{}, {{}, {}, {}}}
```

Comment:
I'm just wodering if k shape is always scalar, why do we need to define it in the test params? Suggest removing this shape from the param list, it will simplify test parameters initialization and test description. And yes it is better to parametrize different k values for different target shapes.

Reply:
Shape params for `k` input has been removed, thank you!
Different `k` values for different target shapes has been already implemented